ddtrace/tracer/textmap.go

```diff
@@ -683,8 +683,9 @@ var (
 	// semi-colon (reserved for separator between entries in the dd list-member),
 	// equals (reserved for list-member key-value separator),
 	// and characters outside the ASCII range 0x21 to 0x7E.
-	// Disallowed characters must be replaced with the underscore.
-	originRgx = regexp.MustCompile(",|=|;|[^\\x21-\\x7E]+")
+	// Disallowed characters must be replaced with the underscore,
+	// except for equals which will be replaced with a tilde
```

Comment:
Just thinking about this more, does this mean we also need to update the extraction side for these trace-state headers? or do we already handle the `~` correctly and convert it back to an equals? @dianashevchenko perhaps?

Reply:
Yeah, I also noticed that. Evan will update this, and I will double-check with system-tests

Reply:
Yep, I think it might require changing [this line](https://github.com/DataDog/dd-trace-go/blob/63174b91c298bd05e3c8af927af530d3b1809b7f/ddtrace/tracer/textmap.go#L879) to replace `~` to `=`